src/storage/v2/durability/snapshot.cpp

```diff
@@ -116,6 +117,13 @@ struct BatchInfo {
 
 // Function used to read information about the snapshot file.
 SnapshotInfo ReadSnapshotInfo(const std::filesystem::path &path) {
+  // Check if snapshot file can be opened and have proper ownership.
+  utils::InputFile file;
+  if (!file.Open(path)) {
```

Comment:
The `exists()` check here is not necessary in the end, since we are doing it in the step before, but checking for access is much cleaner via `access()`.  Moved everything behind `file.cpp` and `file.hpp` utility.